Add annotate to OpenShift DSL

coreydaley opened the pull request:
Adds openshift.selector("foo").annotate([k1:"v1", k2:"v2"], "--overwrite")

Fixes #83

coreydaley commented:
@gabemontero ptal

coreydaley commented:
[test]

coreydaley commented:
[test]

coreydaley commented:
[test]

coreydaley commented:
[test]

coreydaley commented:
[test]

gabemontero commented:
ok cool @coreydaley - comment out the test per our process (we'll uncomment out once the new version of the plugin has been cut), squash the commits, and I'll merge

thanks

gabemontero commented:
cool thanks @coreydaley 

per our bluejeans yesterday, I want to merge the nasty multi-project bug first ... you'll need to rebase this off of that, and then I'll post the merge comment

coreydaley commented:
@gabemontero sounds good

coreydaley commented:
@gabemontero 
As per our conversation on IRC:
 - Rebased PR to reconcile conflicts with the help.jelly file
 - Updated the test to clean up after itself so that it can be run twice in a row
 - Commits have all been squashed

gabemontero commented:
[merge]
